shared/AppInsightsCommon/src/ThrottleMgr.ts

```diff
@@ -181,8 +196,15 @@ export class ThrottleMgr {
                 let interval = config.interval;
                 let monthExpand = (curDate.getUTCFullYear() - date.getUTCFullYear()) * 12 + curDate.getUTCMonth() - date.getUTCMonth();
                 let monthCheck = _checkInterval(interval.monthInterval, 0, monthExpand);
-                let dayCheck = _checkInterval(interval.dayInterval, 0, curDate.getUTCDate()) -1;
-                return monthCheck >= 0 && dayCheck >= 0 && dayCheck <= config.interval.maxTimesPerMonth;
+                
+                let dayCheck = -1;
+                if (_isSpecificDaysGiven) {
+                    dayCheck = arrIndexOf(interval.daysOfMonth, curDate.getUTCDate());
+                } else {
+                    let dayInternal = _checkInterval(interval.dayInterval, date.getUTCDate(), curDate.getUTCDate());
```

Comment:
`getUTCDate()` returns the date of the month, so it's not really the "interval", for example if the previous date was the 31st and today is the 2nd with an interval of 2 should cause it to fire.

I think you need to calculate the number of days between the 2 dates (if there is a last date)
Something like.
let diffDays = Math.floor(curDate.getTime() - date.getTime() / 86400000);

Reply:
I have extra flag `_ isTriggered` to make sure that we only trigger throttle one time a day, but I don't know if this flag could resolve the issue😊